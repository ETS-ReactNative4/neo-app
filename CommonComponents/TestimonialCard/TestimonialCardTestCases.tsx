import React from "react";
import {
  responsiveWidth
  // @ts-ignore
} from "react-native-responsive-dimensions";

import { ITestCase } from "../../TypeInterfaces/TestCases/ITestCases";
import { ScrollView, Alert, View, StyleSheet } from "react-native";
import TestimonialCard from "./TestimonialCard";

const data: ICardData[] = [
  {
    name: "Mihir ",
    image:
      "https://pyt-images.imgix.net/images/testimonials/731/fbProfile/800xh/profile.jpg",
    fallbackImage: "https://pyt-images.imgix.net/images/city/2400xh/phuket.jpg",
    reviewText:
      "PYT are experts in customer satisfaction. they leave no stone unturned.",
    tripType: "Honeymoon",
    region: "Thailand",
    date: "Mar 2019"
  },
  {
    name: "Aarthi",
    image:
      "https://pyt-images.imgix.net/images/testimonials/1022/fbProfile/800xh/profile.jpg",
    fallbackImage: "https://pyt-images.imgix.net/images/city/2400xh/phuket.jpg",
    reviewText: "Tried PYT for the first time and got a well-planned vacation!",
    tripType: "Honeymoon",
    region: "Thailand",
    date: "May 2019"
  },
  {
    name: "Barani",
    image:
      "https://pyt-images.imgix.net/images/testimonials/1073/fbProfile/800xh/profile.jpg",
    fallbackImage: "https://pyt-images.imgix.net/images/city/2400xh/phuket.jpg",
    reviewText:
      "A happy bunch of people who create long-lasting memories with their support",
    tripType: "Honeymoon",
    region: "Thailand",
    date: "Apr 2019"
  }
];

interface ICardData {
  name: string;
  image: string;
  fallbackImage: string;
  reviewText: string;
  tripType: string;
  region: string;
  date: string;
}

interface TestimonialCardWrapperProps {
  cardData: ICardData[];
}

const styles = StyleSheet.create({
  wrapper: {
    height: 184,
    width: responsiveWidth(100)
  }
});

const TestimonialCardWrapper = ({ cardData }: TestimonialCardWrapperProps) => {
  return (
    <View style={styles.wrapper}>
      <ScrollView horizontal showsHorizontalScrollIndicator={false}>
        {cardData.map((dataObj, index) => {
          return (
            <TestimonialCard
              key={index}
              name={dataObj.name}
              image={{ uri: `${dataObj.image}` }}
              fallbackImage={{ uri: `${dataObj.fallbackImage}` }}
              reviewText={dataObj.reviewText}
              tripType={dataObj.tripType}
              region={dataObj.region}
              date={dataObj.date}
              action={() => Alert.alert(`Click -> ${dataObj.name}`)}
            />
          );
        })}
      </ScrollView>
    </View>
  );
};

const TestimonialCardTestCases: ITestCase[] = [
  {
    title: "Testimonial Card",
    Component: (
      <TestimonialCard
        name={"Niranjan Roy"}
        image={{
          uri: "https://pyt-images.imgix.net/images/misc/t21.jpg"
        }}
        fallbackImage={{
          uri: "https://pyt-images.imgix.net/images/city/2400xh/phuket.jpg"
        }}
        reviewText={"These guys sure know how to make their customers happy!"}
        tripType={"Family"}
        region={"Vietnam"}
        date={"Apr 2018"}
        action={() => {}}
      />
    )
  },
  {
    title: "Multiple Testimonial Card",
    Component: <TestimonialCardWrapper cardData={data} />
  }
];

export default TestimonialCardTestCases;
