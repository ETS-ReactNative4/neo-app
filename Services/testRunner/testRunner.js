import renderer from "react-test-renderer";

const testRunner = testCase => {
  test(testCase.title, () => {
    const tree = renderer.create(testCase.Component).toJSON();
    expect(tree).toMatchSnapshot();
  });
};

export default testRunner;
