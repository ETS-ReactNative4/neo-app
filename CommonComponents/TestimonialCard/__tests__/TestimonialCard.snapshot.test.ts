import TestimonialCardTestCases from "../TestimonialCardTestCases";
import testRunner from "../../../Services/testRunner/testRunner";

TestimonialCardTestCases.map(testRunner);
