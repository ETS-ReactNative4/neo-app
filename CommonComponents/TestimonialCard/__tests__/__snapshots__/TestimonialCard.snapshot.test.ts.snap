// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Multiple Testimonial Card 1`] = `
<View
  style={
    Object {
      "height": 184,
      "width": 750,
    }
  }
>
  <ScrollView
    horizontal={true}
    showsHorizontalScrollIndicator={false}
  >
    <TouchableOpacity
      activeOpacity={0.8}
      onPress={[Function]}
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "flex-end",
              "backgroundColor": "rgb(247, 247, 247)",
              "borderColor": "rgba(204, 204, 204, 1)",
              "borderRadius": 8,
              "borderWidth": 1,
              "height": 184,
              "justifyContent": "flex-end",
              "marginBottom": 24,
              "marginRight": 12,
              "paddingLeft": 24,
              "paddingRight": 40,
              "paddingVertical": 32,
              "width": 718,
            },
            undefined,
          ]
        }
      >
        <Image
          onError={[Function]}
          resizeMode="cover"
          source={
            Object {
              "uri": "https://pyt-images.imgix.net/images/testimonials/731/fbProfile/800xh/profile.jpg",
            }
          }
          style={
            Object {
              "borderBottomRightRadius": 125,
              "borderTopLeftRadius": 8,
              "height": 128,
              "left": 0,
              "position": "absolute",
              "top": 0,
              "width": 120,
            }
          }
        />
        <View>
          <Text
            ellipsizeMode="tail"
            numberOfLines={3}
            style={
              Object {
                "color": "rgba(119, 68, 228, 1)",
                "fontFamily": "SourceSansPro-SemiBold",
                "fontSize": 18,
                "lineHeight": 24,
                "marginBottom": 12,
                "textShadowColor": "rgb(247, 247, 247)",
                "textShadowOffset": Object {
                  "height": 1,
                  "width": 1,
                },
                "textShadowRadius": 1,
                "width": 210,
              }
            }
          >
            PYT are experts in customer satisfaction. they leave no stone unturned.
          </Text>
          <Text
            style={
              Object {
                "color": "rgba(44, 47, 66, 1)",
                "fontFamily": "SourceSansPro-SemiBold",
                "fontSize": 13,
                "lineHeight": 13,
                "marginBottom": 4,
              }
            }
          >
            Mihir 
          </Text>
          <Text
            style={
              Object {
                "color": "rgba(83, 87, 109, 1)",
                "fontFamily": "SourceSansPro-SemiBold",
                "fontSize": 13,
                "lineHeight": 13,
              }
            }
          >
            Honeymoon
             trip to 
            Thailand
             in 
            Mar 2019
          </Text>
        </View>
      </View>
    </TouchableOpacity>
    <TouchableOpacity
      activeOpacity={0.8}
      onPress={[Function]}
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "flex-end",
              "backgroundColor": "rgb(247, 247, 247)",
              "borderColor": "rgba(204, 204, 204, 1)",
              "borderRadius": 8,
              "borderWidth": 1,
              "height": 184,
              "justifyContent": "flex-end",
              "marginBottom": 24,
              "marginRight": 12,
              "paddingLeft": 24,
              "paddingRight": 40,
              "paddingVertical": 32,
              "width": 718,
            },
            undefined,
          ]
        }
      >
        <Image
          onError={[Function]}
          resizeMode="cover"
          source={
            Object {
              "uri": "https://pyt-images.imgix.net/images/testimonials/1022/fbProfile/800xh/profile.jpg",
            }
          }
          style={
            Object {
              "borderBottomRightRadius": 125,
              "borderTopLeftRadius": 8,
              "height": 128,
              "left": 0,
              "position": "absolute",
              "top": 0,
              "width": 120,
            }
          }
        />
        <View>
          <Text
            ellipsizeMode="tail"
            numberOfLines={3}
            style={
              Object {
                "color": "rgba(119, 68, 228, 1)",
                "fontFamily": "SourceSansPro-SemiBold",
                "fontSize": 18,
                "lineHeight": 24,
                "marginBottom": 12,
                "textShadowColor": "rgb(247, 247, 247)",
                "textShadowOffset": Object {
                  "height": 1,
                  "width": 1,
                },
                "textShadowRadius": 1,
                "width": 210,
              }
            }
          >
            Tried PYT for the first time and got a well-planned vacation!
          </Text>
          <Text
            style={
              Object {
                "color": "rgba(44, 47, 66, 1)",
                "fontFamily": "SourceSansPro-SemiBold",
                "fontSize": 13,
                "lineHeight": 13,
                "marginBottom": 4,
              }
            }
          >
            Aarthi
          </Text>
          <Text
            style={
              Object {
                "color": "rgba(83, 87, 109, 1)",
                "fontFamily": "SourceSansPro-SemiBold",
                "fontSize": 13,
                "lineHeight": 13,
              }
            }
          >
            Honeymoon
             trip to 
            Thailand
             in 
            May 2019
          </Text>
        </View>
      </View>
    </TouchableOpacity>
    <TouchableOpacity
      activeOpacity={0.8}
      onPress={[Function]}
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "flex-end",
              "backgroundColor": "rgb(247, 247, 247)",
              "borderColor": "rgba(204, 204, 204, 1)",
              "borderRadius": 8,
              "borderWidth": 1,
              "height": 184,
              "justifyContent": "flex-end",
              "marginBottom": 24,
              "marginRight": 12,
              "paddingLeft": 24,
              "paddingRight": 40,
              "paddingVertical": 32,
              "width": 718,
            },
            undefined,
          ]
        }
      >
        <Image
          onError={[Function]}
          resizeMode="cover"
          source={
            Object {
              "uri": "https://pyt-images.imgix.net/images/testimonials/1073/fbProfile/800xh/profile.jpg",
            }
          }
          style={
            Object {
              "borderBottomRightRadius": 125,
              "borderTopLeftRadius": 8,
              "height": 128,
              "left": 0,
              "position": "absolute",
              "top": 0,
              "width": 120,
            }
          }
        />
        <View>
          <Text
            ellipsizeMode="tail"
            numberOfLines={3}
            style={
              Object {
                "color": "rgba(119, 68, 228, 1)",
                "fontFamily": "SourceSansPro-SemiBold",
                "fontSize": 18,
                "lineHeight": 24,
                "marginBottom": 12,
                "textShadowColor": "rgb(247, 247, 247)",
                "textShadowOffset": Object {
                  "height": 1,
                  "width": 1,
                },
                "textShadowRadius": 1,
                "width": 210,
              }
            }
          >
            A happy bunch of people who create long-lasting memories with their support
          </Text>
          <Text
            style={
              Object {
                "color": "rgba(44, 47, 66, 1)",
                "fontFamily": "SourceSansPro-SemiBold",
                "fontSize": 13,
                "lineHeight": 13,
                "marginBottom": 4,
              }
            }
          >
            Barani
          </Text>
          <Text
            style={
              Object {
                "color": "rgba(83, 87, 109, 1)",
                "fontFamily": "SourceSansPro-SemiBold",
                "fontSize": 13,
                "lineHeight": 13,
              }
            }
          >
            Honeymoon
             trip to 
            Thailand
             in 
            Apr 2019
          </Text>
        </View>
      </View>
    </TouchableOpacity>
  </ScrollView>
</View>
`;

exports[`Testimonial Card 1`] = `
<TouchableOpacity
  activeOpacity={0.8}
  onPress={[Function]}
>
  <View
    style={
      Array [
        Object {
          "alignItems": "flex-end",
          "backgroundColor": "rgb(247, 247, 247)",
          "borderColor": "rgba(204, 204, 204, 1)",
          "borderRadius": 8,
          "borderWidth": 1,
          "height": 184,
          "justifyContent": "flex-end",
          "marginBottom": 24,
          "marginRight": 12,
          "paddingLeft": 24,
          "paddingRight": 40,
          "paddingVertical": 32,
          "width": 718,
        },
        undefined,
      ]
    }
  >
    <Image
      onError={[Function]}
      resizeMode="cover"
      source={
        Object {
          "uri": "https://pyt-images.imgix.net/images/misc/t21.jpg",
        }
      }
      style={
        Object {
          "borderBottomRightRadius": 125,
          "borderTopLeftRadius": 8,
          "height": 128,
          "left": 0,
          "position": "absolute",
          "top": 0,
          "width": 120,
        }
      }
    />
    <View>
      <Text
        ellipsizeMode="tail"
        numberOfLines={3}
        style={
          Object {
            "color": "rgba(119, 68, 228, 1)",
            "fontFamily": "SourceSansPro-SemiBold",
            "fontSize": 18,
            "lineHeight": 24,
            "marginBottom": 12,
            "textShadowColor": "rgb(247, 247, 247)",
            "textShadowOffset": Object {
              "height": 1,
              "width": 1,
            },
            "textShadowRadius": 1,
            "width": 210,
          }
        }
      >
        These guys sure know how to make their customers happy!
      </Text>
      <Text
        style={
          Object {
            "color": "rgba(44, 47, 66, 1)",
            "fontFamily": "SourceSansPro-SemiBold",
            "fontSize": 13,
            "lineHeight": 13,
            "marginBottom": 4,
          }
        }
      >
        Niranjan Roy
      </Text>
      <Text
        style={
          Object {
            "color": "rgba(83, 87, 109, 1)",
            "fontFamily": "SourceSansPro-SemiBold",
            "fontSize": 13,
            "lineHeight": 13,
          }
        }
      >
        Family
         trip to 
        Vietnam
         in 
        Apr 2018
      </Text>
    </View>
  </View>
</TouchableOpacity>
`;
